File: spring-boot-starter-breuninger-validation/src/main/java/com/breuninger/boot/validation/validators/InstantValidator.java
package com.breuninger.boot.validation.validators;

import java.time.Instant;
import java.time.format.DateTimeParseException;

import javax.validation.ConstraintValidator;
import javax.validation.ConstraintValidatorContext;

public class InstantValidator implements ConstraintValidator<IsInstant, String> {

  @Override
  public void initialize(final IsInstant constraintAnnotation) {
  }

  @Override
  public boolean isValid(final String value, final ConstraintValidatorContext context) {
    if (value == null) {
      return true;
    }
    try {
      Instant.parse(value);
    } catch (final DateTimeParseException e) {
      return false;
    }
    return true;
  }
}

File: validation/src/main/kotlin/com/breuninger/boot/validation/constraintvalidators/InstantValidator.kt
package com.breuninger.boot.validation.constraintvalidators

import com.breuninger.boot.validation.constraints.IsInstant
import java.time.Instant
import java.time.format.DateTimeParseException
import javax.validation.ConstraintValidator
import javax.validation.ConstraintValidatorContext

class InstantValidator : ConstraintValidator<IsInstant, String> {

  override fun initialize(constraintAnnotation: IsInstant?) {
  }

  override fun isValid(value: String?, context: ConstraintValidatorContext): Boolean {
    return value?.let {
      return try {
        Instant.parse(value)
        true
      } catch (dateTimeParseException: DateTimeParseException) {
        false
      }
    } ?: true
  }
}
